### ISSUE TITLE
Inappropriate Error Message

### ISSUE THREAD

**Issue by vipinbose** [symfony/symfony#36382]
**Symfony version(s) affected**:  5.0.*

**Description**  
I was migrating from 4.4.3 to 5.0.0 and found some error message misleading like the following

Argument 1 passed to App\EventListener\SessionListener::onKernelRequest() must be an instance of Symfony\Component\HttpKernel\Event\GetResponseEvent, instance of Symfony\Component\HttpKernel\Event\RequestEvent given,

The same was the case with 
GetResponseForExceptionEvent against 

**How to reproduce**  

Just swap Objects 

**Possible Solution**  

Argument 1 passed to App\EventListener\SessionListener::onKernelRequest() must be an instance of Symfony\Component\HttpKernel\Event\RequestEvent, instance of Symfony\Component\HttpKernel\Event\GetResponseEvent given,

**Additional context**

**Comment by fancyweb** [comment]
Where did you see this error?

**Comment by vipinbose** [comment]
I am getting this in a Listener class. I was migrating it from 4.4 to 5.+

Argument 1 passed to App\EventListener\SessionListener::onKernelRequest() must be an instance of Symfony\Component\HttpKernel\Event\GetResponseEvent, instance of Symfony\Component\HttpKernel\Event\RequestEvent given, called in .........................../vendor/symfony/event-dispatcher/Debug/WrappedListener.php on line 117

The error gone away when I replaced GetResponseEvent with RequestEvent [Opposite to what is mentioned in the error message]

 public function onKernelRequest(GetResponseEvent $event)  - Error

 public function onKernelRequest(RequestEvent $event) - No Error

**Comment by fancyweb** [comment]
Status: works for me

I think I understand it. The message is actually a PHP `\TypeError`.

In Symfony 4.4, we deprecated `GetResponseEvent` in favor of `RequestEvent`. In Symfony 5.0, we removed `GetResponseEvent`.

Your listener method signature uses `GetResponseEvent`. In Symfony 5.0, we now pass an instance of `RequestEvent` to the listener, hence the `\TypeError`. What you did to make it work is the correct solution, ie stop using the deprecated class and use the new one.
